sdk/go/common/workspace/creds.go

```diff
@@ -213,5 +213,23 @@ func StoreCredentials(creds Credentials) error {
 	if err != nil {
 		return errors.Wrapf(err, "marshalling credentials object")
 	}
-	return ioutil.WriteFile(credsFile, raw, 0600)
+
+	tempCredsFile, err := ioutil.TempFile("", "credentials-*.json")
```

Comment:
Looks like Windows is failing with:

> problem logging in: rename C:\Users\RUNNER~1\AppData\Local\Temp\credentials-502826475.json D:\a\pulumi\pulumi\src\stackName-1606979904380190300\credentials.json: The system cannot move the file to a different disk drive.

One workaround would be to create the temp file in the same dir as the final file. The downside is leftover temp files may be left around in error cases, but that should hopefully be rare. If we do this, should we try to remove the temp file if the rename fails?